### ISSUE TITLE
`non_constant_identifier_names` false positives for pattern fields

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#4299]
Follow-up from #4297.

Repro:

```dart
f(Object o) {
  switch(o) {
    case (:int AB, var b):
  }
}
```
